### ISSUE TITLE
Twitter login fail

### ISSUE THREAD

**Issue by mvolfik** [cgeo/cgeo#7130]
##### Detailed steps causing the problem:
- Settings
- Services
- Social media --> Twitter
-  - [x] Activate
- Authorize c:geo
- Start authorization

##### Actual behavior after performing these steps:
Short loading _"Waiting for Twitter"_ and then shows "toast" message _"c:geo failed to initialize authorization process"_

##### Is the problem reproducible for you?
- [x] Yes

##### System information:
```
--- System information ---
Device: LG-D405 (w7n_global_com, lge)
Android version: 5.0.2
Android build: LRX22G.A1454843669
c:geo version: 2018.08.14
Google Play services: enabled - 12.8.74 (020304-204998136)
Low power mode: inactive
Compass capabilities: yes
Rotation vector sensor: present
Orientation sensor: present
Magnetometer & Accelerometer sensor: present
Direction sensor used: rotation vector
Hide own/found: false
Map strategy: auto
HW acceleration: enabled (default state)
System language: cs_CZ (c:geo forced to English)
System date format: M/d/y
Debug mode active: no
System internal c:geo dir: /data/data/cgeo.geocaching (274,4 MB free) internal
User storage c:geo dir: /storage/emulated/0/cgeo (274,4 MB free) external non-removable
Geocache data: /storage/emulated/0/Android/data/cgeo.geocaching/files/GeocacheData (274,4 MB free) external non-removable
Database: /data/data/cgeo.geocaching/databases/data (1,1 MB) on system internal storage
Fine location permission: granted
Write external storage permission: granted
Geocaching sites enabled:
   geocaching.com: Logged in (Login OK) / BASIC
Geocaching.com date format: dd.MM.yyyy
Installed c:geo plugins:  none
--- End of system information ---
```

**Comment by Lineflyer** [comment]
Reproducible for me.
I receive a 403 as result of the token request.

**Comment by Lineflyer** [comment]
https://twittercommunity.com/t/callback-url-not-approved-but-worked-for-months/107261

**Comment by Lineflyer** [comment]
@mucek4 Who has access data for the Twitter developer account we use? Seems it is not our normal `@android_gc` account.

**Comment by mucek4** [comment]
I have no idea. Let's apply for new key with our twitter account.

**Comment by Lineflyer** [comment]
After inquiring @mucek4 the current developer account at Twitter is most likely not any longer known. Therefore I applied for a new developer access using our "normal" twitter profile. Waiting for feedback now.

**Comment by Lineflyer** [comment]
Don't ask me why, but access to development console is now working for `@android_gc`. Seems that the profile was somehow broken and repaired itself by renewing the application.

The current configured callback URL is `http://www.cgeo.org`. As far as I can see we are using `callback://wwww.cgeo.org/twitter/` in c:geo. However this URL can not be configured at Twitter as it is no valid URL due to the protocol prefix not being http. Seems change in the app is needed anyway.

**Comment by rsudev** [comment]
I briefly checked the twitter dev docs:

    Mobile apps with app-specific protocols must use just the protocol
    Example:
    
    You want to use example://authorize as your callback URL
        Add this to both your Twitter app dashboard and your call to oauth/request_token: example://

Could you try to enter just 'callback://' there and than retry authorization?

**Comment by Lineflyer** [comment]
Will try tonight and report back.

**Comment by Lineflyer** [comment]
Good catch @rsudev 
I changed the configuration on Twitter to ``callback://`` and it is working again. Tested with 2018.08.28-NB.

@mvolfik Can you please check and report back whether it is also working for you now?

**Comment by mvolfik** [comment]
Sorry for replying so late, but I just confirm it works for me.

Dne 30. 8. 2018 10:59 napsal uživatel "Lars" <notifications@github.com>:

Closed #7130 <https://github.com/cgeo/cgeo/issues/7130>.

—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub
<https://github.com/cgeo/cgeo/issues/7130#event-1817641345>, or mute the
thread
